src/object.d

```diff
@@ -4795,3 +4795,152 @@ TTo[] __ArrayCast(TFrom, TTo)(TFrom[] from) @nogc pure @trusted
     foreach (v; s)
         assert(v == cast(short) 0xabab);
 }
+
+// Yields the appropriate printf format token for a type T
+// Indended to be used by miniFormat
+private template getPrintfFormat(T)
+{
+    static if (is(T == long))
+    {
+        enum getPrintfFormat = "%lld";
+    }
+    else static if (is(T == ulong))
+    {
+        enum getPrintfFormat = "%llu";
+    }
+    else static if (__traits(isIntegral, T))
+    {
+        static if (__traits(isUnsigned, T))
+        {
+            enum getPrintfFormat = "%u";
+        }
+        else
+        {
+            enum getPrintfFormat = "%d";
+        }
+    }
+    else
+    {
+        static assert(0, "Unknown format");
+    }
+}
+
+/**
+Minimalistic formatting for use in _d_assert_fail to keep the compilation
+overhead small and avoid the use of Phobos.
+*/
+private auto miniFormat(V)(auto ref V v)
+{
+    import core.stdc.stdio : sprintf;
+    import core.stdc.string : strlen;
+    static if (__traits(isIntegral, V))
+    {
+        enum printfFormat = getPrintfFormat!V;
+        char[20] val;
+        sprintf(val.ptr, printfFormat, v);
+        return val.idup[0 .. strlen(val.ptr)];
+    }
+    else static if (__traits(isFloating, V))
```

Comment:
Why a separate case for floats, why not handle this in `getPrintfFormat`?

Reply:
Because I kept the formatting very naive for floating points numbers and always use `%g` and a very wide buffer for the reasons mentioned above (first version of this, happens before program termination).

Reply:
So you're thinking this will be extended with a more complicated formatting of floats?